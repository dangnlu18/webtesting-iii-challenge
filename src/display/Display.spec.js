// Test away!
import React from 'react';
import * as rtl from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Display from './Display';



test('Renders open/close and locked/unlocked', ()=>{
	const wrapper = rtl.render(< Display/>);

	expect(wrapper.asFragment()).toMatchSnapshot();

})

test('Displays CLOSED if closed prop is true', async ()=>{
	const wrapper = rtl.render(< Display closed={true}/>);

	const closed = wrapper.getByText(/closed/i);
	expect(closed).toHaveTextContent('Closed');

	
})


test('When closed, uses red-led class', ()=>{
	const wrapper = rtl.render(< Display closed={true}/>);

	const closed = wrapper.getByText(/closed/i);
	expect(closed).toHaveClass('red-led');
})

test('When unlocked, uses green-led class', ()=>{
	const wrapper = rtl.render(< Display closed={true}/>);

	const unlocked = wrapper.getByText(/unlocked/i);
	expect(unlocked).toHaveClass('green-led');
})

test('Initial state is unlocked', async ()=>{
	const wrapper = rtl.render(< Display/>);
	await wrapper.findByText(/unlocked/i);

	wrapper.debug();
})

test('Initial state is open', async ()=>{
	const wrapper = rtl.render(< Display/>);
	await wrapper.findByText(/open/i);

	wrapper.debug();
})