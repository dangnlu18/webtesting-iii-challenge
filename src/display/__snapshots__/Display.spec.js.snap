// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders open/close and locked/unlocked 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;
